src/utilities.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * getJSON attempts to read the given file. If successful,
+ * it returns an object of the contained JSON. If the extraction
+ * fails, it returns an empty object.
+ */
+export async function getJSON(file: string) {
+  try {
+    return JSON.parse(await Deno.readTextFile(file));
+  } catch (_) {
+    return {};
```

Comment:
I mentioned this in a previous comment, but looking back at this, I think in the situation where an error is raised here (could be because: file not found, no permissions to the file, or the file is corrupted and not valid JSON anymore), the reason for these errors is useful context to the end-user and thus we should bubble it up. I realize that since this method is also used in `check-update`, this may require tweaking elsewhere, but IMO that is a worthwhile effort. My 2 cents.

Reply:
I agree with you, but I'd really like to punt on this for this PR if you're OK with it. Refactoring this is something we marked as a to do with the `check-update` hook, as well, and I'm a little nervous to re-write the functionality of that file along with this new feature. I'm happy to put in a TODO and circle back to it personally.

Reply:
That sounds fine